org.testeditor.web.dropwizard.xtext/src/main/java/org/testeditor/web/xtext/index/persistence/GradleProjectIndexUpdater.xtend

```diff
@@ -0,0 +1,78 @@
+package org.testeditor.web.xtext.index.persistence
+
+import com.google.common.io.CharStreams
+import java.io.File
+import java.io.InputStreamReader
+import java.nio.charset.StandardCharsets
+import java.nio.file.Files
+import java.nio.file.Paths
+import java.nio.file.StandardOpenOption
+import java.util.concurrent.TimeUnit
+import javax.inject.Inject
+import org.testeditor.web.xtext.index.CustomStandaloneBuilder
+import javax.inject.Singleton
+import org.slf4j.LoggerFactory
+
+@Singleton
+class GradleProjectIndexUpdater extends IndexUpdater {
+	
+	static val logger = LoggerFactory.getLogger(GradleProjectIndexUpdater)
+
+	@Inject CustomStandaloneBuilder builder
+	
+	static val GRADLE_PROCESS_TIMEOUT = 10 -> TimeUnit.MINUTES
+
+	override def void initIndex(File file) {
+		builder.languages = languageAccessors
+		if (new File(file.absolutePath, 'build.gradle').exists) {
+			runGradleAssemble(file)
+			prepareGradleTask(file)
+			val jars = collectClasspathJarsViaGradle(file)
+			builder => [
+				baseDir = file.absolutePath
+				sourceDirs = #[file.absolutePath + "/src/main/java", file.absolutePath + '/src/test/java']
+				classPathEntries = jars + #[file.absolutePath + '/build/classes/java/main']
+			]
+			builder.launch // does all the indexing ...
+		} else {
+			super.initIndex(file)
+		}
+	}
+
+	/** make sure the task 'printTestClasspath' exists */
+	private def void prepareGradleTask(File repoRoot) {
+		val process = new ProcessBuilder('./gradlew', 'tasks', '--all').directory(repoRoot).start
+		logger.info('running gradle tasks.')
+		process.waitFor(GradleProjectIndexUpdater.GRADLE_PROCESS_TIMEOUT.key, GradleProjectIndexUpdater.GRADLE_PROCESS_TIMEOUT.value) // allow for downloads and the like
```

Comment:
qualifier `GradleProjectIndexUpdater` shouldn't be necessary here, right?
(two more occurrences of this further down)

Reply:
damn renaming xtend stuff, always adds full qualified names. I'll clean that up.